FirebaseAuth/Sources/User/FIRUser.m

```diff
@@ -583,6 +590,108 @@ - (void)setTokenService:(FIRSecureTokenService *)tokenService
+                                        *_Nullable request,
+                                    NSError *_Nullable error))completion {
+  FIRAuthRequestConfiguration *requestConfiguration = self->_auth.requestConfiguration;
+
+  FIRStartPasskeyEnrollmentRequest *request =
+      [[FIRStartPasskeyEnrollmentRequest alloc] initWithIDToken:self.rawAccessToken
+                                           requestConfiguration:requestConfiguration];
+  [FIRAuthBackend
+      startPasskeyEnrollment:request
+                    callback:^(FIRStartPasskeyEnrollmentResponse *_Nullable response,
+                               NSError *_Nullable error) {
+                      if (error) {
+                        completion(nil, error);
+                        return;
+                      } else {
+                        // cached the passkey name.  This is needed when calling
+                        // finalizePasskeyEnrollment
+                        self.passkeyName = name;
+                        NSData *challengeInData =
+                            [[NSData alloc] initWithBase64EncodedString:response.challenge
+                                                                options:0];
+                        NSData *userIdInData =
+                            [[NSData alloc] initWithBase64EncodedString:response.userID options:0];
+
+                        ASAuthorizationPlatformPublicKeyCredentialProvider *provider =
+                            [[ASAuthorizationPlatformPublicKeyCredentialProvider alloc]
+                                initWithRelyingPartyIdentifier:response.rpID];
+                        ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest *request =
+                            [provider
+                                createCredentialRegistrationRequestWithChallenge:challengeInData
+                                                                            name:name
+                                                                          userID:userIdInData];
+                        completion(request, nil);
+                      }
+                    }];
+}
+
+- (void)finalizePasskeyEnrollmentWithPlatformCredential:
+            (ASAuthorizationPlatformPublicKeyCredentialRegistration *)platformCredential
+                                             completion:(nullable void (^)(
+                                                            FIRAuthDataResult *_Nullable authResult,
+                                                            NSError *_Nullable error))completion {
+  dispatch_async(FIRAuthGlobalWorkQueue(), ^{
+    FIRAuthDataResultCallback decoratedCallback =
+        [FIRAuth.auth signInFlowAuthDataResultCallbackByDecoratingCallback:completion];
+    FIRAuthRequestConfiguration *requestConfiguration = self->_auth.requestConfiguration;
+
+    NSString *credentialID = [platformCredential.credentialID base64EncodedStringWithOptions:0];
```

Comment:
What does 0 mean here? Is there a more semantic option?

Reply:
Copied the options available here.
```
/****************        Base 64 Options	****************/

typedef NS_OPTIONS(NSUInteger, NSDataBase64EncodingOptions) {
    // Use zero or one of the following to control the maximum line length after which a line ending is inserted. No line endings are inserted by default.
    NSDataBase64Encoding64CharacterLineLength = 1UL << 0,
    NSDataBase64Encoding76CharacterLineLength = 1UL << 1,

    // Use zero or more of the following to specify which kind of line ending is inserted. The default line ending is CR LF.
    NSDataBase64EncodingEndLineWithCarriageReturn = 1UL << 4,
    NSDataBase64EncodingEndLineWithLineFeed = 1UL << 5,

} API_AVAILABLE(macos(10.9), ios(7.0), watchos(2.0), tvos(9.0));
```

Reply:
Got it, it's bit-based options so `0` should be fine.